### ISSUE TITLE
RCS1151 with default interface implementation

### ISSUE THREAD

**Issue by bddckr** [JosefPihrt/Roslynator#588]
**Product and Version Used**: `Roslynator.Analyzers`, version `2.1.0`.

**Steps to Reproduce**:
1. Create an interface with any member (of any type, e.g. method or property, etc.) and add a default implementation for that member.
1. Create a new type that implements that interface.
1. Create an instance of that type and try to call the interface member on it by casting to the interface first.

**Actual Behavior**:
```csharp
public interface ITestInterface
{
    public void TestMethod() => Console.WriteLine("Test");
}

public readonly struct TestStruct : ITestInterface
{
}
```
And somewhere else try to call that member (the method in the example) on any instance of the type that implements the interface:
```c#
// RCS1151	Remove redundant cast.
((ITestInterface)default(TestStruct)).TestMethod();
```

**Expected Behavior**:
Since the compiler forces us to cast to the interface, RCS1151 should not raise any diagnostic for default interface implementations of any member kind.

There is a statement in the docs, explaining the need for the cast:
>That cast from `SampleCustomer` to `ICustomer` is necessary. The `SampleCustomer` class doesn't need to provide an implementation for `ComputeLoyaltyDiscount`; that's provided by the `ICustomer` interface. However, the `SampleCustomer` class doesn't inherit members from its interfaces. That rule hasn't changed. In order to call any method declared and implemented in the interface, the variable must be the type of the interface, `ICustomer` in this example.
>https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/default-interface-members-versions#upgrade-with-default-interface-members
